### PR TITLE
Move the plotting section to map docs

### PR DESCRIPTION
**Pull request by hayesla**
This PR addresses #6866, which moves the plotting docs into the map tutorial section, and removes the plotting section (which included a matplotlib intro). The reader is now referred to the matplotlib docs for more information on this

### CODE REVIEW

File: docs/intro/maps.rst [hunk 1]
```diff
@@ -195,25 +195,63 @@ Plotting Maps
...
+    import sunpy.map
+    import sunpy.data.sample
+    aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+    aia_map.peek()
 
-* :ref:`sphx_glr_generated_gallery_plotting_wcsaxes_plotting_example.py`
 
-* :ref:`sphx_glr_generated_gallery_plotting_map_editcolormap.py`
+This creates a plot window with all axes defined, a plot title, and the image of the map data defined by the contents of the map.
+In non-interactive mode the plot window blocks the command line terminal and must be closed before doing anything else.
```
Comment (dstansby) [inline docs/intro/maps.rst:219]:
```suggestion
In non-interactive mode the plot window blocks execution and must be closed before doing anything else.
```
This then covers running from a script too

**Review by dstansby — APPROVED** [review]
If CI is happy, I'm happy 😄 . I think lots of this content needs updating, but for this PR just moving it is an improvement 👍

**Comment by hayesla** [comment]
Yeah I totally agree, the content needs updating. I'll open an issue about this now

File: docs/tutorial/maps.rst [hunk 2]
```diff
@@ -324,6 +362,156 @@ Note how the colorbar does not change since these two plots share the same color
...
+
+Check out the following foundational examples in the Example Gallery for plotting Maps:
+
+* :ref:`sphx_glr_generated_gallery_plotting_aia_example.py`
+
+* :ref:`sphx_glr_generated_gallery_plotting_wcsaxes_plotting_example.py`
+
+* :ref:`sphx_glr_generated_gallery_plotting_map_editcolormap.py`
+
+* :ref:`sphx_glr_generated_gallery_plotting_grid_plotting.py`
```
[the remark this answers is not in the document]
Comment (ayshih) [reply docs/tutorial/maps.rst:450]: As pointed out by @alasdairwilson, see sphinx-gallery/sphinx-gallery#1029

**Review by nabobalis — APPROVED** [review]

**Pull request merged**